### PR TITLE
[BREAKING] Remove the legacy AST JSON output

### LINKED ISSUE #9512: Remove support for the old "legacy" JSON AST
The compiler has two JSON AST outputs: the "legacy" and the "compact" form. They mostly contain the same detail but in a vastly different structure.

For a long time Remix static analyzer was relying on the legacy format, but this is not the case anymore. We have also discussed this during our call with frameworks.

*I could not find an issue for this, despite our numerous discussions about it.*

### PR DESCRIPTION
**Pull request by axic**
Fixes #9512.

This can be squashed prior to merge.

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Breaking Changes:
  * Code Generator: Use ``revert`` with error signature ``Panic(uint256)`` and error codes instead of invalid opcode on failing assertions.
  * Type System: Explicit conversions from literals to integer type is as strict as implicit conversions.
  * Type System: Explicit conversions from literals to enums are only allowed if the value fits in the enum.
+ * Command Line Interface: Remove the legacy ``--ast-json`` option. Only the ``--ast-compact-json`` option is supported now.
+ * Standard JSON: Remove the ``legacyAST`` option.
```
Comment (axic) [inline Changelog.md]: I did not sort these, as they will be sorted before the release anyway.

File: test/libsolidity/ASTJSONTest.h [hunk 2]
```diff
@@ -64,13 +65,11 @@ class ASTJSONTest: public TestCase
 	) const;
 
 	std::vector<std::pair<std::string, std::string>> m_sources;
-	std::string m_expectationLegacy;
 	std::string m_expectationParseOnly;
 	std::string m_astFilename;
 	std::string m_astParseOnlyFilename;
 	std::string m_legacyAstFilename;
```
Comment (Marenz) [inline test/libsolidity/ASTJSONTest.h]: This can also be removed, right?

**Review by Marenz — CHANGES_REQUESTED** [review]

File: Changelog.md (hunk 1 above)
Comment (mijovic) [reply Changelog.md]: I guess it is fine, but it would be great to keep it as sorted as possible, so it is less work for release :)

**Comment by mijovic** [comment]
Looks good apart from what @Marenz already requested

File: Changelog.md (hunk 1 above)
Comment (axic) [reply Changelog.md]: I kept it sorted on other PRs but it quickly fell out of being sorted with other PR so I don't bother with it anymore :)

Comment (mijovic) [reply Changelog.md]: Fair enough

File: test/libsolidity/ASTJSONTest.h (hunk 2 above)
Comment (axic) [reply test/libsolidity/ASTJSONTest.h]: Done.

**Comment by axic** [comment]
Should I squash this?

**Review by chriseth — APPROVED** [review]

**Pull request merged**